bandc_c+a.rb

```diff
@@ -1,27 +1,43 @@
 puts "Welcome! Select classic or advanced game! For classic press 'c', for advanced press 'a'."
 game_type = gets.chomp
 
-while game_type == "c" or "a"
+def classic_check(variable)
+  if variable.size != 4
+    puts "Please, try again. Your guess must contain exactly 4 digits!"
+  end
+        
+  if variable.uniq.length != 4
+    puts "Please, try again. Your guess must contain 4 unique digits!"
+  end
+    
+  if variable.include?(0)
+    puts "Please, try again. You cannot use zero in your guess!"
+  end
+end
+
+def advanced_check(variable)
+  if variable.size != 6
+    puts "Please, try again. Your guess must contain exactly 6 digits!"
+  end
+        
+  if variable.uniq.length != 6
+    puts "Please, try again. Your guess must contain 6 unique digits!"
+  end
+end
+
+while game_type == "c" or "a" do
   if game_type == "c"  
     secret_number = Array(1..9).sample(4)
+    p secret_number #delete after testing
     classic_game = [0, 1, 2, 3]
     user_guess = ""
     
+    puts "Input your guess using keyboard."
+    
     while user_guess != secret_number do
-      puts "Input your guess using keyboard."
       user_guess = gets.chomp.split("").map { |el| el.to_i }
       
-      if user_guess.size != 4
-        puts "Please, try again. Your guess must contain exactly 4 digits!"
-      end
-        
-      if user_guess.uniq.length != 4
-        puts "Please, try again. Your guess must contain 4 unique digits!"
-      end
-    
-      if user_guess.include?(0)
-        puts "Please, try again. You cannot use zero in your guess!"
-      end
+      classic_check(user_guess)
```

Comment:
it's really just for checking if there is any mistake in user's input. if there is, it prints some messages. :D